x/merkledb/db.go

```diff
@@ -1166,12 +1167,11 @@ func (db *merkleDB) invalidateChildrenExcept(exception *view) {
 // Otherwise leave [db.root] as Nothing.
 func (db *merkleDB) initializeRoot() error {
```

Comment:
The bug here was that this function assumes that the DB is in a consistent state but we called it prior to repairing the DB.